src/bun.js/bindings/bindings.cpp

```diff
@@ -3172,6 +3189,33 @@ JSC__JSValue JSC__JSValue__getIfPropertyExistsImpl(JSC__JSValue JSValue0,
     return JSC::JSValue::encode(object->getIfPropertyExists(globalObject, property));
 }
 
+extern "C" JSC__JSValue JSC__JSValue__getIfPropertyExistsImplString(JSC__JSValue JSValue0, JSC__JSGlobalObject* globalObject, BunString* propertyName)
+{
+    JSValue value = JSC::JSValue::decode(JSValue0);
+    if (UNLIKELY(!value.isObject()))
+        return JSValue::encode({});
+
+    JSC::VM& vm = globalObject->vm();
+    JSC::JSObject* object = value.getObject();
+    auto identifier = JSC::Identifier::fromString(vm, propertyName->toWTFString(BunString::ZeroCopy));
+    auto property = JSC::PropertyName(identifier);
+
+    return JSC::JSValue::encode(object->getIfPropertyExists(globalObject, property));
+}
+
+extern "C" JSC__JSValue JSC__JSValue__getOwn(JSC__JSValue JSValue0, JSC__JSGlobalObject* globalObject, BunString* propertyName)
+{
+    VM& vm = globalObject->vm();
+    JSValue value = JSC::JSValue::decode(JSValue0);
+    auto identifier = JSC::Identifier::fromString(vm, propertyName->toWTFString(BunString::ZeroCopy));
+    auto property = JSC::PropertyName(identifier);
+    PropertySlot slot(value, PropertySlot::InternalMethodType::GetOwnProperty);
+    if (value.getOwnPropertySlot(globalObject, property, slot)) {
+        return JSValue::encode(slot.getValue(globalObject, property));
+    }
+    return JSValue::encode({});
```

Comment:
I don't know how this is valid C++, what does `{}` return, isn't that a block?

Reply:
It's an empty initialization list, that invokes the default initializer for a struct/class fields (https://en.cppreference.com/w/cpp/language/aggregate_initialization)

Reply:
Looks like C++ has changed a bit since I last used it extensively, my JS focus has made me rusty in native code. Thanks for the explanations, much appreciated!